### PR TITLE
trace: provide appName and version from Config

### LINKED ISSUE #1883: Modify `trace` to be used by Hypersdk
Regarding the `trace` package. Hypersdk needs to uses something very similar for VMs tracing (https://github.com/ava-labs/hypersdk/pull/330).
To use the existing package in a more generic way, we need to get some constants from the Config.
- `AppName`
- `Version`

### PR DESCRIPTION
**Pull request by najeal**
## Why this should be merged

This resolves #1883 

## How this works

This deletes the usage of constants in the `trace` package. Values are provided from the Config.

### CODE REVIEW

File: trace/tracer.go [hunk 1]
```diff
@@ -36,6 +33,9 @@ type Config struct {
 	// If >= 1 always samples.
 	// If <= 0 never samples.
 	TraceSampleRate float64 `json:"traceSampleRate"`
+
+	AppName string
+	Version string
```
Comment (danlaine) [inline trace/tracer.go:38]: Can you add JSON tags for these?

**Review by danlaine — COMMENTED** [review]
Thanks for the contribution!

File: trace/tracer.go (hunk 1 above)
Comment (najeal) [reply trace/tracer.go]: Done.
We need to marshal the config at some point?

Comment (danlaine) [reply trace/tracer.go]: It can be marshaled in the admin API's `GetConfig` method

**Review by danlaine — APPROVED** [review]

File: config/config.go [hunk 2]
```diff
@@ -1225,6 +1226,7 @@ func getTraceConfig(v *viper.Viper) (trace.Config, error) {
 	if !enabled {
 		return trace.Config{
 			Enabled: false,
+			AppName: constants.AppName,
```
Comment (StephenButtolph) [inline config/config.go:1229]:
```suggestion
```

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**